### PR TITLE
Build tree-widget when using plists

### PR DESCRIPTION
**Pull request by fredericgiquel**
Possible solution for #3497 

With this PR, capabilities are displayed in 'lsp-session' buffer both with plist and hash-table.

### CODE REVIEW

File: lsp-protocol.el [hunk 1]
```diff
@@ -260,7 +260,8 @@ Allowed params: %s" interface (reverse (-map #'cl-first params)))
       (defalias 'lsp-copy 'copy-sequence)
       (defun lsp-member? (from key)
         (when (listp from)
-          (plist-member from key))))
+          (plist-member from key)))
+      (defalias 'lsp-structure-p 'json-plist-p))
```
Comment (kiennq) [inline lsp-protocol.el:264]: Is using `plistp` enough? Or you may have to include `(require 'json)` for this.

**Review by yyoncho — APPROVED** [review]

File: lsp-protocol.el (hunk 1 above)
Comment (fredericgiquel) [reply lsp-protocol.el:264]: AFAIK, `plistp` does not exist. So I will include the require.

Comment (kiennq) [reply lsp-protocol.el:265]: It's in the latest Emacs snapshot :)
https://github.com/emacs-mirror/emacs/commit/dedd19a2f5cbf16c8ac8a122b0c39ee4e178b9e8
Anyway, using `json-plist-p` is fine.

**Review by kiennq — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**